internal/common/logs.go

```diff
@@ -50,15 +53,19 @@ func writeToLogs(filePath string, duration time.Duration, logLinesPerMinute int)
 	defer ticker.Stop()
 	endTimeout := time.After(duration)
 
-	//loop until the test duration is reached
+	// Sending the logs within the first minute before the ticker kicks in the next minute
+	for i := 0; i < logLinesPerMinute; i++ {
+		_, err := f.WriteString(fmt.Sprintf(logLine, i))
```

Comment:
The first sending would be more prioritized in catching error. If it fails during the first time, then we would return error immediately  without worrying about [the second write](https://github.com/aws/amazon-cloudwatch-agent-test/pull/165/files#diff-32ea2f52d0d817b42a874abb6e550c6c2040c34707954b8efd66e0c474660738R68). However, if the first write success, then it would guarantee to have no errors in the second write.

Reply:
I don't understand. What second write are you referring to? The loop below only has one WriteString call.

Reply:
So we have two writes:
* First write will be in the first minute
```
for i := 0; i < logLinesPerMinute; i++ {
		_, err := f.WriteString(fmt.Sprintf(logLine, i))
```
* Second write would be after 60 second
```
for {
		select {
		case <-ticker.C:
			for i := 0; i < logLinesPerMinute; i++ {
				_, err := f.WriteString(fmt.Sprintf("# %d - This is a log line.", i))
				if err != nil {
					log.Printf("Error in writing a string to the file %s: %v ", filePath, err)
				}
```
So we only need to catch the first error instead of catching the second write errors